### PR TITLE
Improve diagnostics E0425: `use` (public) name

### PR DESCRIPTION
**Pull request by liigo**
E0425: unresolved name

### CODE REVIEW

File: src/librustc_resolve/diagnostics.rs [hunk 1]
```diff
@@ -843,6 +843,15 @@ Or:
 let unknown_variable = 12u32;
 let x = unknown_variable; // ok!
 ```
+
+Please verify that the name is declared by `use` item and ensure it's public
```
Comment (GuillaumeGomez) [inline src/librustc_resolve/diagnostics.rs]: I'd say: "If the item isn't available in the current module, please check you imported it:".

File: src/librustc_resolve/diagnostics.rs [hunk 2]
```diff
@@ -843,6 +843,15 @@ Or:
...
+to us:
+
+```
```
Comment (GuillaumeGomez) [inline src/librustc_resolve/diagnostics.rs]: It'll fail because `foo::bar` doesn't exist. Add `ignore` flag.

**Comment by liigo** [comment]
@GuillaumeGomez Thank you! I've updated (and rebased).

File: src/librustc_resolve/diagnostics.rs [hunk 3]
```diff
@@ -843,6 +843,16 @@ Or:
...
 ```
+
+If the item isn't available in the current module, please check you imported it:
+
+```ignore
+use foo::bar;
+bar();
+```
+
+...and check the item is public, non-private at least, to the current module.
```
Comment (GuillaumeGomez) [inline src/librustc_resolve/diagnostics.rs]: "... and check if the item is public, non-private at least, to the current module."

Comment (liigo) [inline src/librustc_resolve/diagnostics.rs]: https://is.gd/8CFKu9

Comment (GuillaumeGomez) [inline src/librustc_resolve/diagnostics.rs]: It was just because you forgot a whitespace after "...".

**Comment by liigo** [comment]
updated (and rebased)

File: src/librustc_resolve/diagnostics.rs [hunk 4]
```diff
@@ -843,6 +843,17 @@ Or:
...
+If the item isn't available in the current module, please check you imported it:
+
+```ignore
+use foo::bar;
+bar();
+```
+
+... and check if the item is public, non-private at least, to the current
+module.
+
```
Comment (GuillaumeGomez) [inline src/librustc_resolve/diagnostics.rs]: And this is an extra empty line. :laughing:

File: src/librustc_resolve/diagnostics.rs [hunk 5]
```diff
@@ -843,6 +843,17 @@ Or:
 let unknown_variable = 12u32;
 let x = unknown_variable; // ok!
 ```
+
+If the item isn't available in the current module, please check you imported it:
```
Comment (nikomatsakis) [inline src/librustc_resolve/diagnostics.rs]: I think I would word this slightly differently:

"If the item is not defined in the current module, it must be imported using a `use` statement, like so:"

File: src/librustc_resolve/diagnostics.rs [hunk 6]
```diff
@@ -843,6 +843,17 @@ Or:
...
+
+```ignore
+use foo::bar;
+bar();
+```
+
+... and check if the item is public, non-private at least, to the current
```
Comment (nikomatsakis) [inline src/librustc_resolve/diagnostics.rs]: Maybe: "If the item you are importing is not defined in some super-module of the current module, then it must also be declared as public (e.g., `pub fn`)."

**Comment by liigo** [comment]
@nikomatsakis Thank you! Done.

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**